config_db.go

```diff
@@ -133,18 +136,97 @@ func byIDs(ids []string) bson.M {
 	return bson.M{idKey: bson.M{"$in": ids}}
 }
 
-func getSectionsBSON(ctx context.Context, ids []string) ([]bson.Raw, error) {
-	cur, err := GetEnvironment().DB().Collection(ConfigCollection).Find(ctx, byIDs(ids))
+func getSectionsBSON(ctx context.Context, ids []string, includeOverrides bool) ([]bson.Raw, error) {
+	missingIDs := ids
+	docs := make([]bson.Raw, 0, len(ids))
+	if includeOverrides {
+		cur, err := GetEnvironment().DB().Collection(ConfigCollection).Find(ctx, byIDs(ids))
+		if err != nil {
+			return nil, errors.Wrap(err, "finding local configuration sections")
+		}
+
+		if err := cur.All(ctx, &docs); err != nil {
+			return nil, errors.Wrap(err, "iterating cursor for local configuration sections")
+		}
+
+		var docIDs []string
+		for _, doc := range docs {
+			id, err := doc.LookupErr("_id")
+			if err != nil {
+				continue
+			}
+			idString, ok := id.StringValueOK()
+			if !ok {
+				continue
+			}
+			docIDs = append(docIDs, idString)
+		}
+
+		missingIDs, _ = utility.StringSliceSymmetricDifference(ids, docIDs)
+	}
+
+	if GetEnvironment().SharedDB() == nil || len(missingIDs) == 0 {
+		return docs, nil
+	}
+
+	cur, err := GetEnvironment().SharedDB().Collection(ConfigCollection).Find(ctx, byIDs(missingIDs))
```

Comment:
If I'm reading this right, does this mean editing just 1 field locally would change the whole section and not get it from shared db?

Reply:
yes! that's the idea. I'll try to make it clearer with comments. 
If any document exists in the local db it would completely override the shared version.

Reply:
done